examples/pose-apriltag/rs-pose-apriltag.cpp

```diff
@@ -0,0 +1,296 @@
+            auto tf_fisheye_to_tag = tags.pose_in_camera[t];
+            auto tf_world_to_body = to_transform(camera_world_pose.rotation, camera_world_pose.translation);
+            tags.pose_in_world[t] = tf_world_to_body * tf_body_to_fisheye * tf_fisheye_to_tag;
+        }
+    }
+    
+    static void undistort(apriltag_detection_t& src, const rs2_intrinsics& intr) {
+        deproject(src.c, intr, src.c);
+        
+        double corr_arr[4][4];
+        for(int c=0; c<4; ++c){
+            deproject(src.p[c], intr, src.p[c]);
+            
+            corr_arr[c][0] = (c==0 || c==3) ? -1 : 1;
+            corr_arr[c][1] = (c==0 || c==1) ? -1 : 1;
+            corr_arr[c][2] = src.p[c][0];
+            corr_arr[c][3] = src.p[c][1];
+        }
+        matd_destroy(src.H);
+        src.H = homography_compute2(corr_arr);
+    }
+    
+    static void deproject(double pt[2], const rs2_intrinsics& intr, const double px[2]) {
+        float fpt[3], fpx[2] = { (float)px[0], (float)px[1] };
+        rs2_deproject_pixel_to_point(fpt, &intr, fpx, 1.0f);
+        pt[0] = fpt[0];
+        pt[1] = fpt[1];
+    }
+};
+
+int main(int argc, char * argv[]) try
+{
+    // Declare RealSense pipeline, encapsulating the actual device and sensors
+    rs2::pipeline pipe;
+    // Create a configuration for configuring the pipeline with a non default profile
+    rs2::config cfg;
+    // Add pose stream
+    cfg.enable_stream(RS2_STREAM_POSE, RS2_FORMAT_6DOF);
+    // Enable both image streams
+    // Note: It is not currently possible to enable only one
+    cfg.enable_stream(RS2_STREAM_FISHEYE, 1, RS2_FORMAT_Y8);
+    cfg.enable_stream(RS2_STREAM_FISHEYE, 2, RS2_FORMAT_Y8);
+    
+    // replay
+    if(argc > 1){ cfg.enable_device_from_file(argv[1]); }
+    
+    // Start pipe and get camera calibrations
+    const int fisheye_sensor_idx = 1;  //for the left fisheye lens of T265
+    auto pipe_profile       = pipe.start(cfg);
+    auto fisheye_stream     = pipe_profile.get_stream(RS2_STREAM_FISHEYE, fisheye_sensor_idx);
+    auto fisheye_intrinsics = fisheye_stream.as<rs2::video_stream_profile>().get_intrinsics();
+    auto body_fisheye_extr  = fisheye_stream.get_extrinsics_to(pipe_profile.get_stream(RS2_STREAM_POSE));
+
+    // Create an Apriltag detection manager
+    apriltag_manager tag_manager(fisheye_intrinsics, body_fisheye_extr);
+    
+    // Main loop
+    while (true)
+    {
+        // Wait for the next set of frames from the camera
+        auto frames        = pipe.wait_for_frames();
+        auto fisheye_frame = frames.get_fisheye_frame(fisheye_sensor_idx);
+        auto frame_number  = fisheye_frame.get_frame_number();
+        auto camera_pose   = frames.get_pose_frame().get_pose_data();
+        
+        if(frame_number % 4 == 0)
+        {
+            fisheye_frame.keep();
+            std::async(std::launch::async, [img=fisheye_frame,fn=frame_number,pose=camera_pose,&tag_manager](){
```

Comment:
Initialization in lambda captures is a C++14 feature. You may need to stick to C++11 unless it is ok for libRealSense to change the standard in a single sample.

Reply:
updated to use `std::bind()`